src/engine/Minus.cpp

```diff
@@ -0,0 +1,234 @@
+      result = a;
+      *dynResult = result.moveToDynamic();
+      return;
+    } else {
+      // This case should never happen.
+      AD_CHECK(false);
+    }
+  }
+
+  IdTableStatic<A_WIDTH> a = dynA.asStaticView<A_WIDTH>();
+  IdTableStatic<B_WIDTH> b = dynB.asStaticView<B_WIDTH>();
+  IdTableStatic<OUT_WIDTH> result = dynResult->moveToStatic<OUT_WIDTH>();
+
+  std::vector<size_t> rightToLeftCols(b.cols(),
+                                      std::numeric_limits<size_t>::max());
+  for (const auto& jc : joinColumns) {
+    rightToLeftCols[jc[1]] = jc[0];
+  }
+
+  size_t ia = 0, ib = 0;
+  while (ia < a.size() && ib < b.size()) {
+    // Join columns 0 are the primary sort columns
+    while (a(ia, joinColumns[0][0]) < b(ib, joinColumns[0][1])) {
+      // Write a result
+      result.emplace_back();
+      size_t backIdx = result.size() - 1;
+      for (size_t col = 0; col < a.cols(); col++) {
+        result(backIdx, col) = a(ia, col);
+      }
+      ia++;
+      if (ia >= a.size()) {
+        goto finish;
+      }
+    }
+    while (b(ib, joinColumns[0][1]) < a(ia, joinColumns[0][0])) {
+      ib++;
+      if (ib >= b.size()) {
+        goto finish;
+      }
+    }
+
+    while (b(ib, joinColumns[0][1]) == a(ia, joinColumns[0][0])) {
+      // check if the rest of the join columns also match
+      RowComparison rowEq = isRowEqSkipFirst(a, b, ia, ib, joinColumns);
+      switch (rowEq) {
+        case RowComparison::EQUAL: {
+          ia++;
+          if (ia >= a.size()) {
+            goto finish;
+          }
+        } break;
+        case RowComparison::NOT_EQUAL_LEFT_SMALLER: {
+          // ib does not discard ia, and there can not be another ib that
+          // would discard ia.
+          result.emplace_back();
+          size_t backIdx = result.size() - 1;
+          for (size_t col = 0; col < a.cols(); col++) {
+            result(backIdx, col) = a(ia, col);
+          }
+          ia++;
+          if (ia >= a.size()) {
+            goto finish;
+          }
+        } break;
+        case RowComparison::NOT_EQUAL_RIGHT_SMALLER: {
+          ib++;
+          if (ib >= b.size()) {
+            goto finish;
+          }
+        } break;
+        default:
+          AD_CHECK(false);
+      }
+    }
+  }
+finish:
+  result.reserve(result.size() + (a.size() - ia));
+  while (ia < a.size()) {
+    result.emplace_back();
+    size_t backIdx = result.size() - 1;
+    for (size_t col = 0; col < a.cols(); col++) {
+      result(backIdx, col) = a(ia, col);
+    }
+    ia++;
+  }
+  *dynResult = result.moveToDynamic();
+}
+
+template <int A_WIDTH, int B_WIDTH>
+Minus::RowComparison Minus::isRowEqSkipFirst(
+    const IdTableStatic<A_WIDTH>& a, const IdTableStatic<B_WIDTH>& b, size_t ia,
+    size_t ib, const vector<array<size_t, 2>>& joinColumns) {
+  for (size_t i = 1; i < joinColumns.size(); ++i) {
+    Id va = a(ia, joinColumns[i][0]);
+    Id vb = b(ib, joinColumns[i][1]);
```

Comment:
For which of the types do you want me to use auto?
I personally am not a huge fan of auto, as I feel like it makes the code harder to read. I'd rather use type aliases, which can have a descriptive name while still allowing for easy type changes (such as the `Id` type).

Reply:
Ok, my argument was more for easiness of refactoring, but possibly it is also good to once look at all the changes
you have to do when changing such stuff.

Could I interest you in the uniform initialization syntax
`Id va{a(ia, col[i[0])}` 
Then we would have all the benefits, since we also get errors for narrowing conversions.
(I will also try to use this in the future :))

Reply:
@joka921 I'm not a fan of the curly brace initialization, as it is a rather unusual syntax, but given that it has clear advantages in this case I'll switch over to it.